scala/scala.bzl

```diff
@@ -494,6 +511,55 @@ def _scala_test_impl(ctx):
     _write_test_launcher(ctx, rjars)
     return _scala_binary_common(ctx, cjars, rjars)
 
+def _prep_grep_pattern_from(patterns):
+    combined_pattern = ""
+    for pattern in patterns:
+      combined_pattern += " -e {}\.class".format(pattern)
+    return combined_pattern
+
+def _discover_classes(ctx, patterns, archive):
+    discovered_classes = ctx.new_file(ctx.label.name + "_discovered_classes.txt")
+    ctx.action(
+      inputs=[archive],
+      outputs=[discovered_classes],
+      progress_message="Discovering classes with patterns of %s" % patterns,
+      #TODO consider with Damien/Ulf/Oscar the implications of switching from grep to scala code
+      #Pro-> logic will be cohesive (currently the scala code assumes stuff from the grep)
+      #Con-> IIRC Ulf warned me about performance implications of these traversals
+      command="unzip -l {archive} | grep {combined_patterns} > {out}".format(archive=archive.path, combined_patterns=_prep_grep_pattern_from(patterns), out=discovered_classes.path))
```

Comment:
I moved to jar but when I tried to use the one from bazel_tools I got the following error:
```
____Loading package: test ____Loading package: @local_config_cc// ____Loading package: @local_config_xcode// ____Loading complete. Analyzing... ____Loading package: @io_bazel_rules_scala_org_specs2_specs2_matcher//jar ____Loading package: @io_bazel_rules_scala_org_openjdk_jmh_jmh_generator_asm//jar ____Loading package: @io_bazel_rules_scala_org_openjdk_jmh_jmh_generator_reflection//jar ____Found 89 targets... ____Building... ____[342 / 391] Discovering classes with patterns of ["E2E", "IT"] ERROR: /Users/ittaiz/workspace/rules_scala/test/BUILD:245:1: error executing shell command: 'external/local_jdk/bin/jar -tf bazel-out/local-fastbuild/bin/test/JunitMultipleSuffixes.jar | grep -e E2E\.class -e IT\.class > bazel-out/local-fastbuild/bin/test/JunitMultipleSuffixes_discovered_...' failed: Process exited with status 1 [sandboxed]. Error: could not find libjava.dylib Error: Could not find Java SE Runtime Environment. Use --strategy=Generating=standalone to disable sandboxing for the failing actions. ____Building complete. ____Elapsed time: 3.011s, Critical Path: 0.66s
```
My attempt was via adding an internal attribute (like _ijar), add it to the inputs of the action and then use it instead of the vanilla jar command but that doesn't seem like it's enough. It seems more stuff are needed as inputs. Any idea?